### ISSUE TITLE
Restic restore use preexisting files

### ISSUE THREAD

**Issue by JensRantil** [restic/restic#3337]
Output of `restic version`
--------------------------
I don't have it installed.

What should restic do differently? Which functionality do you think we should add?
----------------------------------------------------------------------------------
When restoring multiple files where network is the bottleneck it would be really useful to be able to point restic to a set of files/directories which it can use to check if files to restore already exists. If so, it copies those files locally, instead of downloading from a remote repository.

What are you trying to do? What problem would this solve?
---------------------------------------------------------
I am trying to speed up restoring my backup by using a pool of already existing files on my computer. More specifically, I'm looking into using restic for Cassandra sstable backups. When restoring a Cassandra snapshot, many of them will likely exists locally.

Did restic help you today? Did it make you happy in any way?
------------------------------------------------------------
I am not a user of restic but I'm excited about its deduplication functionality.

**Comment by MichaelEischer** [comment]
Would #407 solve your problem?

**Comment by JensRantil** [comment]
> Would #407 solve your problem?

No, not really. The preexisting files might reside in a different directory than what I'm restoring to.
